### ISSUE TITLE
[resize-observer] The initial size of ResizeObservation

### ISSUE THREAD

**Issue by cathiechen** [w3c/csswg-drafts#3664]
https://drafts.csswg.org/resize-observer/#dom-resizeobservation-resizeobservation-target-options

> 4. Set this internal lastReportedSize slot to (0,0)

Not very strong opinion about this.
We might miss this:
- Element's size isn't (0, 0)
- ro.observe(Element)
- Element's size change to (0, 0)

**Comment by cathiechen** [comment]
Add a case to reproduce this issue.
https://jsbin.com/qotevalowe/1/edit?html,console,output

**Comment by mrego** [comment]
> Add a case to reproduce this issue.
> https://jsbin.com/qotevalowe/1/edit?html,console,output

In that example if you remove the last line (the one that changes the size to 0x0), then you'll get notified for a size change of 100x100 (when nothing has actually changed). I guess it's because the initial `lastReportedSize` is 0x0 and it detects a change.

**Comment by cathiechen** [comment]
@mrego Yes, exactly like you said. :)  We might also get notified when nothing has changed.

**Comment by atotic** [comment]
There was a discussion about what initial size should be at:
https://github.com/WICG/ResizeObserver/issues/8

Options were:
* current size of the element
  - rejected because this might cause RO.observe() to trigger layout.
* (0,0)
  - causes RO to fire immediately after being observed, unless Element size is 0,0
  - 0,0 is also Element's size when hidden. RO will not fire initially until Element is shown.
- undefined
  - would cause RO to always fire, even when hidden.

I think (0,0) option is most useful, because it does not fire until Element is shown.

**Comment by Loirooriol** [comment]
This is inconsistent with `IntersectionObserver`, which always delivers a notification just after observing, since

 1. When [observing a target Element](https://w3c.github.io/IntersectionObserver/#observe-target-element), `previousThresholdIndex` is set to -1.
 2. When [running the update intersection observations steps](https://w3c.github.io/IntersectionObserver/#update-intersection-observations-algo), `thresholdIndex` is set to a non-negative index:
    > Set *thresholdIndex* to the index of the first entry in *observer*.[`thresholds`](https://w3c.github.io/IntersectionObserver/#dom-intersectionobserver-thresholds) whose value is greater than *intersectionRatio*, or the length of *observer*.[`thresholds`](https://w3c.github.io/IntersectionObserver/#dom-intersectionobserver-thresholds) if *intersectionRatio* is greater than or equal to the last entry in *observer*.[`thresholds`](https://w3c.github.io/IntersectionObserver/#dom-intersectionobserver-thresholds).
 3. This ensures a IntersectionObserverEntry will be queued:
    > If *thresholdIndex* does not equal *previousThresholdIndex* or if *isIntersecting* does not equal *previousIsIntersecting*, [queue an IntersectionObserverEntry](https://w3c.github.io/IntersectionObserver/#queue-an-intersectionobserverentry)

Similarly, I think that [`ResizeObservation`](https://drafts.csswg.org/resize-observer/#dom-resizeobservation-resizeobservation-target-options) should initialize `[lastReportedSizes](https://drafts.csswg.org/resize-observer/#dom-resizeobservation-lastreportedsizes)` to something invalid like `[(-1,-1)]`, so that we consistently get a notification regardless of the current size.

`contain-intrinsic-size: auto` also needs something like this, it uses an internal ResizeObserver to store the [last remembered size](https://drafts.csswg.org/css-sizing-4/#last-remembered). And 0x0 is a valid size that should still be stored, so the callback should be invoked. See https://github.com/w3c/csswg-drafts/issues/7539

> I think (0,0) option is most useful, because it does not fire until Element is shown.

I don't think this justifies not reporting an observation for elements that do have a box but it just happens to be 0x0. Also note IntersectionObserver notifies about disconnected elements:

```js
new IntersectionObserver(console.log).observe(document.createElement("div"));
```

**Comment by emilio** [comment]
Agenda+ to confirm the resolution in https://github.com/w3c/csswg-drafts/issues/7539#issuecomment-1202608735 applies to ResizeObserver too (or get a new one otherwise), since from discussion with @Loirooriol it wasn't quite clear from the minutes.
